### DISCUSSION TITLE
Use `slug` for `fields.image` filename in `fields.array` when avaliable

### DISCUSSION THREAD

**Discussion by tordans** [Thinkmill/keystatic#1307] [Ideas]
When `fields.image` is used inside a `fields.array` object the path to the image key that is used for the image folder path is the index of the array.

Keystatic is smart enough to also move the images and update the data when the array order is changed.

However, this results in git commits where images are replace over and over whenever the order is changed.

This could be prevented, if the image folder path where to use something else than the array index.

One option would be, to use the array object slug field when such a field is present in the config.

https://keystatic.com/docs/fields/image
https://docsmill.dev/npm/@keystatic/core@0.5.34#/.fields.image

**Comment by emmatown** [comment]
This exists with the `slugField` option on arrays: https://keystatic.com/docs/fields/array#slug-field

**Reply by tordans** [reply]
I looked into this again, today and found that for file uploads I cannot change the file name, just the file path, is this correct?

- Before `/authors/0/file.pdf`
- After `/authors/<slug>/file.pdf`

Is there a way to change this to `/authors/<slug>.pdf` ?

Update: I worked around this using the `download` property on the link:

```

                <a
                  href={file.file}
                  download={file.name.slug}
                >
                  {file.name.name}
                </a>
```

… which will set the file name in the browser.
